pom.xml

```diff
@@ -154,7 +154,7 @@
                             replace="[[${filename}_\2]]${line.separator}\1 \2" flags="g" />
 
                           <!-- fix includes to contain asciidoc suffix (what is omitted on github wiki) -->
-                          <replaceregex pattern="include\:\:(.*)\[" replace="include::\1.asciidoc["
+                          <replaceregex pattern="include\:\:(.*?)\.(?!cfg|xml|sql|properties|csv|json)\[\]" replace="include::\1.asciidoc["
```

Comment:
Excellent catch with the lookahead and inverse check. I was not aware that you can actually do it like this with regex. Have you tested this with e.g. regex plugin in Eclipse? I am not sure if the `!` will apply to all the following OR-ed terms.
Further, you are have added an additional closing bracket at the end (`\]`). This is IMHO a bug. Please update this PR and also please test this change as we want to avoid releasing this to maven central, update all documentation POMs in all repos just to notice that our generation is now entirely broken ;)

Reply:
I would even suggest that we should implement a small github actions build to have proper CI CD in place. It does not make sense to spend that amount of testing all the time. We could easily provide regression tests by cloning repos like cobigen or devon4j, update docgen version to catch up the build version and then build other repos even in parallel to see whether a PR is breaking this in a regression test.

What do you think?

Reply:
I completely agree with that. But IMHO I think we definitely need a TESTING/STAGING environment to check in real time how the website will be updated before we actually merge it to reflect in the original website. The github actions will run in stage/testing env first.  WDYT?

Reply:
Fully agree. The only question is how to stage possibly multiple PRs in one staging area? I think possibly a zip for downloading for a successful build might be enough to host and try it on your local?.
Anyhow, coming to finalization here. Could you please give it a manual try and test your version with devonfw guide once?

Reply:
Yes ZIP would be really helpful.

Reply:
Actually why would someone include a `*.json` or `*.sql`, ... file from AsciiDoc? Do we really have such cases? Seems odd to me...

Reply:
basically, it could be all filetypes as you can include different file types for source code plotting. I think, that's the reason why the file types are listed here. 
Anyhow, this raises the question, why to no turn around the logic to have inclusion matching rather than exclusion.
What about 

`pattern="include\:\:(.*)\.(asciidoc|adoc)\[" replace="include::\1.asciidoc["`

Reply:
> Excellent catch with the lookahead and inverse check. I was not aware that you can actually do it like this with regex. Have you tested this with e.g. regex plugin in Eclipse? I am not sure if the `!` will apply to all the following OR-ed terms.
> Further, you are have added an additional closing bracket at the end (`\]`). This is IMHO a bug. Please update this PR and also please test this change as we want to avoid releasing this to maven central, update all documentation POMs in all repos just to notice that our generation is now entirely broken ;)

@hohwille  - Agreed. provided the fix now. tested all the possibilities and works fine. Kindly review again.

Reply:
> basically, it could be all filetypes as you can include different file types for source code plotting. I think, that's the reason why the file types are listed here.
> Anyhow, this raises the question, why to no turn around the logic to have inclusion matching rather than exclusion.
> What about
> 
> `pattern="include\:\:(.*)\.(asciidoc|adoc)\[" replace="include::\1.asciidoc["`

@maybeec  Addressed that in the latest commit.